gems/smithy-schema/lib/smithy-schema/document.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+require_relative 'document_utils'
+
+module Smithy
+  module Schema
+    # A Smithy document type, representing typed or untyped data from Smithy data model.
+    # ## Document types
+    # Document types are protocol-agnostic view of untyped data. They could be combined
+    # with a schema to serialize its contents.
+    #
+    # Smithy-Ruby currently only support JSON documents.
```

Comment:
Not sure if we could use the same class to support XML documents in the future. We could potentially create some extension points from here and change DocumentUtils into XML/JSON specific document builders/parsers. Future problem if there's a use case.